platform/posix/transport/include/openssl_posix.h

```diff
@@ -118,6 +118,11 @@ typedef struct OpensslCredentials
      */
     const char * sniHostName;
 
+    /**
```

Comment:
sniHostName == NULL should result in SNI being disabled rather than an explicit disableHostnameCheck.